src/eckit/parser/YAMLParser.cc

```diff
@@ -425,12 +426,47 @@ Value YAMLParser::parseNumber() {
 }
 
 
```

Comment:
By preference would be to put helper functions into an anonymous namespace. And at the very least they should be declared static so they don't leak into wider linkage.